src/comparator/field_comparator.py

```diff
@@ -260,7 +260,7 @@ def _is_parent_type(
                     child_type
                 )
             )
-        if parent_type not in [parent.type for parent in parent_resources]:
+        if parent_type not in [parent.value.type for parent in parent_resources]:
             # Resulting referenced resource patterns cannot be resolved identical.
```

Comment:
`parent` is a  `WithLocation` object, call `value` attr to get the `type` of the resource.